### ISSUE TITLE
CBLListener readOnly property prevents pulls from updating the database

### ISSUE THREAD

**Issue by brendand** [couchbase/couchbase-lite-ios#1106]
It was recommended setting the readOnly property on CBLListener for improved security so nobody could push any documents to a listener. It seems not only to prevent pushing to the listener, but also the device running the listener is prevented from pulling from another device.

I've based my peer sync code on the Bonjour Grocery Sync example (although in Objective-C instead of Swift). And it works perfectly when the CBLListener readOnly property is set to NO.

Here's some log messages to show what's happening to the puller:

```
Sync: CBLRestPuller[https://iPhone-6s.local.:49743/db-d9b8dd6ae89c48da9637bfba68b35f9a]: Server is CouchbaseLite 1.2 (unofficial)
Sync: CBLRestPuller[https://iPhone-6s.local.:49743/db-d9b8dd6ae89c48da9637bfba68b35f9a]: Replicating from lastSequence=802 
SyncVerbose: CBLRestPuller[https://iPhone-6s.local.:49743/db-d9b8dd6ae89c48da9637bfba68b35f9a] starting ChangeTracker: mode=0, since=802 
SyncVerbose: CBLSocketChangeTracker[0x6100001bfd40 db-d9b8dd6ae89c48da9637bfba68b35f9a]: POST //iPhone-6s.local.:49743/db-d9b8dd6ae89c48da9637bfba68b35f9a/_changes 
CBLSocketChangeTracker[0x6100001bfd40 db-d9b8dd6ae89c48da9637bfba68b35f9a]: Cant connect, giving up: Error Domain=CBLHTTP Code=403 "403 forbidden" UserInfo={NSURL=https://iPhone-6s.local.:49743/db-d9b8dd6ae89c48da9637bfba68b35f9a/_changes, NSLocalizedFailureReason=forbidden, NSLocalizedDescription=403 forbidden} 
Sync: CBLRestPuller[https://iPhone-6s.local.:49743/db-d9b8dd6ae89c48da9637bfba68b35f9a]: ChangeTracker stopped; error=Error Domain=CBLHTTP Code=403 "403 forbidden" UserInfo={NSURL=https://iPhone-6s.local.:49743/db-d9b8dd6ae89c48da9637bfba68b35f9a/_changes, NSLocalizedFailureReason=forbidden, NSLocalizedDescription=403 forbidden} 
Sync: CBLRestPuller[https://iPhone-6s.local.:49743/db-d9b8dd6ae89c48da9637bfba68b35f9a] Progress: set error = 403 forbidden 
Sync: CBLRestPuller[https://iPhone-6s.local.:49743/db-d9b8dd6ae89c48da9637bfba68b35f9a] STOPPING...
```

So basically `Code=403 forbidden` seems to be the issue with pulling from another device with the readOnly property of the listeners set to ON.

Using Couchbase Lite 1.2 GM on iOS 9.3 and Mac OS X 10.11.4.

**Comment by snej** [comment]
The read-only auth check in the listener needs to allow the POST method when the path is `/*/_changes`.

**Comment by brendand** [comment]
I can confirm that there are no more errors syncing with the readOnly property enabled on my CBLListener instances with this patch. Thanks for fixing it Jens!
